src/Microsoft.PowerShell.Commands.Utility/commands/utility/Csv.cs

```diff
@@ -20,92 +22,69 @@ internal CSVHelper(char delimiter)
         /// </summary>
         internal char Delimiter { get; } = ',';
 
+        internal char Quote { get; } = '"';
+
         /// <summary>
         /// Parse a CSV string.
         /// </summary>
         /// <param name="csv">
         /// String to be parsed.
         /// </param>
         internal Collection<string> ParseCsv(string csv)
-        {
+        {        
+            csv = csv.Trim(); 
             Collection<string> result = new Collection<string>();
-            string tempString = string.Empty;
-            csv = csv.Trim();
             if (csv.Length == 0 || csv[0] == '#')
             {
                 return result;
             }
 
-            bool inQuote = false;
-            for (int i = 0; i < csv.Length; i++)
+            var reader = new StringReader(csv);
+            StringBuilder stringBuilder = new StringBuilder();
```

Comment:
I would suggest renaming `stringBuilder`  to something which reflects what is stored in the builder, instead of reflecting what type of object it is. Perhaps something such as `wordBuffer` as you're building up a word and clear it when the word has been read fully?